src/loki/trace_route_action.cc

```diff
@@ -118,15 +118,16 @@ void loki_worker_t::init_trace(valhalla_request_t& request) {
   // we require shape or encoded polyline but we dont know which at first
   if (!request.options.shape_size()) {
     throw valhalla_exception_t{114};
-  };
+  }
 
   // Determine max factor, defaults to 1. This factor is used to increase
   // the max value when an edge_walk shape match is requested
   float max_factor = 1.0f;
-  std::string shape_match =
-      rapidjson::GetValueByPointerWithDefault(request.document, "/shape_match", "walk_or_snap")
-          .GetString();
-  if (shape_match == "edge_walk") {
+  // If shape_match was not specified then set to default of "walk_or_snap"
+  if (!request.options.has_shape_match()) {
+    request.options.set_shape_match("walk_or_snap");
```

Comment:
why use strings. we should make `shape_match` and enum shouldnt we?

Reply:
I thought the same thing yesterday when I was doing the tests. Ran out of time yesterday - made a note for myself. Are you okay if I make another issue and merge this now?

Reply:
yep!